packages/mobile/src/screens/settings-screen/InboxSettingsScreen.tsx

```diff
@@ -64,8 +76,51 @@ const useStyles = makeStyles(({ spacing, palette, typography }) => ({
   }
 }))
 
+const options = [
+  {
+    title: messages.allTitle,
+    description: messages.allDescription,
+    value: ChatPermission.ALL
+  },
+  {
+    title: messages.followeeTitle,
+    description: messages.followeeDescription,
+    value: ChatPermission.FOLLOWEES
+  },
+  {
+    title: messages.tipperTitle,
+    description: messages.tipperDescription,
+    value: ChatPermission.TIPPERS
+  },
+  {
+    title: messages.noneTitle,
+    description: messages.noneDescription,
+    value: ChatPermission.NONE
+  }
+]
+
 export const InboxSettingsScreen = () => {
   const styles = useStyles()
+  const dispatch = useDispatch()
+  const permissions: Record<ID, ValidatedChatPermissions> = useSelector(
+    getUserChatPermissions
+  )
+  const userId = useSelector(getUserId)
+  const currentPermission = userId ? permissions[userId]?.permits : null
+
+  useEffect(() => {
+    if (userId) {
+      dispatch(fetchPermissions({ userIds: [userId] }))
+    }
+  }, [dispatch, userId])
+
+  const handlePress = useCallback(
+    (value) => {
+      dispatch(setPermissions({ permissions: value }))
```

Comment:
Thoughts on not using redux + sagas for the setPermissions stuff?

I skipped it in the web impl. could probably share a hook with ya if we're looking to keep things DRY:

https://github.com/AudiusProject/audius-client/blob/776596ff5f1b1dc09a148d834ab91e90884a6ebc/packages/web/src/components/inbox-settings-modal/InboxSettingsModal.tsx#L97-L116

Reply:
What's the benefit of doing it this way?

Reply:
Mainly just colocation making things easier to reason about. I'm game either way, either I can adopt the new saga or abstract the component methods into a hook you can use

Reply:
I would vote for not having so much async logic directly in the component. I know we don't have a story yet for how to do mutations without sagas. This could at least be moved to a shared state hook of some sort that exports a save callback?

I _am_ in favor of getting off of sagas and potentially skipping redux state management for local component state. Especially for new code.

Reply:
Got it, I think what was missing from my understanding was that this is basically local state. I'll convert to a hook.

Reply:
> This could at least be moved to a shared state hook of some sort that exports a save callback?

yep, precisely